cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/ProgramControllerServiceAdapter.java

```diff
@@ -90,7 +95,10 @@ public void running() {
 
       @Override
       public void failed(Service.State from, Throwable failure) {
+        Throwable rootCause = Throwables.getRootCause(failure);
```

Comment:
why the root cause? Is it possible that you drop important context if you get the root cause?